pennylane/gradients/__init__.py

```diff
@@ -11,7 +11,266 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""This subpackage contains quantum gradient transforms."""
+"""
+
+Quantum gradient transforms are strategies of computing the gradient of a quantum
+circuit by **transforming** the quantum circuit into one or more gradient circuits.
+These gradient circuits, once executed and post-processed, return the gradient
+of the original circuit.
+
+Examples of quantum gradient transforms include finite-differences and parameter-shift
+rules.
+
+This module provides a selection of device-independent, differentiable quantum
+gradient transforms. As such, these quantum gradient transforms can be used to
+compute the gradients of quantum circuits on both simulators and hardware.
+
+In addition, it also includes an API for writing your own quantum gradient
+transforms.
+
+These quantum gradient transforms can be used in two ways:
+
+- Transforming quantum circuits directly
+- Registering a quantum gradient strategy for use when performing autodifferentiation
+  with a :class:`~.QNode`.
+
+Overview
+--------
+
+Gradient transforms
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+    :toctree: api
+
+    finite_diff
+    param_shift
+    param_shift_cv
+
+Custom gradients
+^^^^^^^^^^^^^^^^
+
+.. autosummary::
+    :toctree: api
+
+    gradient_transform
+
+Utility functions
+^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+    :toctree: api
+
+    finite_diff_coeffs
+    generate_shifted_tapes
+    compute_vjp
+    batch_vjp
+    vjp
+
+
+Registering autodifferentiation gradients
+-----------------------------------------
+
+All PennyLane QNode's are automatically differentiable, and can be included
+seamlessly within an autodiff pipeline. When creating a :class:`~.QNode`, the
+strategy for determining the optimal differentiation strategy is *automated*,
+and takes into account the circuit, device, autodiff framework, and metadata
+(such as whether finite shots are used).
+
+.. code-block:: python
+
+    @qml.qnode(dev)

```

Comment:
Perhaps use a more informative code block here that specifies an interface, shots, etc. to match with the previous description.

Reply:
done!